cmake/Reprojucer.cmake

```diff
@@ -2640,11 +2675,13 @@ endfunction()
 
 function(__set_JucePlugin_Build_defines target target_type)
 
-  # See XCodeProjectExporter::Target::getTargetSettings()
-  # in JUCE/extras/Projucer/Source/Project Saving/jucer_ProjectExport_XCode.h
-  set(plugin_types     VST VST3 AudioUnit AudioUnitv3  RTAS AAX Standalone  )
-  set(setting_suffixes VST VST3 AUDIOUNIT AUDIOUNIT_V3 RTAS AAX AUDIOUNIT_V3)
-  set(define_suffixes  VST VST3 AU        AUv3         RTAS AAX Standalone  )
+  if(JUCER_BUILD_AUDIOUNIT_V3 AND DEFINED JUCER_VERSION AND JUCER_VERSION VERSION_LESS 5)
+    set(JUCER_BUILD_STANDALONE_PLUGIN ON)
+  endif()
+
+  set(plugin_types     VST VST3 AudioUnit AudioUnitv3  RTAS AAX Standalone       )
+  set(setting_suffixes VST VST3 AUDIOUNIT AUDIOUNIT_V3 RTAS AAX STANDALONE_PLUGIN)
```

Comment:
What is this doing?

Reply:
These 3 lists (`plugin_types`, `setting_suffixes`, and `define_suffixes`) are used in the following `foreach` loop, where we iterate over them at the same time (with an index going from 0 to 6 included). I could have unrolled that `foreach` loop and written the "same" code 7 times, but using these lists is more compact. The function is used to set the 7 `JucePlugin_Build_*` defines to either 1 or 0. I could also re-write the code to look more like what Projucer does: https://github.com/WeAreROLI/JUCE/blob/master/extras/Projucer/Source/ProjectSaving/jucer_ProjectExporter.cpp#L411-L435 (but not in this PR).

Reply:
And the diff is dealing with the fact that in JUCE 4, the Standalone plugin is depending on the AUv3 one, while it's not the case anymore in JUCE 5.